lasagna-master/lasagna_master.go

```diff
@@ -0,0 +1,36 @@
+package lasagna
+
+func PreparationTime(layers []string, timePerLayer int) int {
+	if timePerLayer == 0 {
+	   timePerLayer = 2
+   }
+   return timePerLayer * len(layers)
+}
+
+func Quantities(layers []string) (noodles int, sauce float64) {
+	noodles = 0
+	sauce = 0.0
+	for _, layer := range layers {
+		if layer == "sauce" {
+			sauce += .2
+		} else if layer == "noodles" {
+			noodles += 50
+		}
+	}
+	return
+}
+
+func AddSecretIngredient(friendsList, myList []string) []string {
+	return append(myList, friendsList[len(friendsList) - 1])
+}
+
+func ScaleRecipe(quantities []float64, amounts int) []float64 {
+	scaledQuantities := make([]float64, len(quantities))
+	for i, quantity := range quantities {
+	 scaledQuantities[i] = quantity * float64(amounts) / 2
```

Comment:
Two comments here. First of all, nice! This is a good way to do it as you avoid `append` capacity allocations. It would need some benchmark though to see what is better performance wise. You can also get rid of the slice you create with make and create an array directly with `[len(quantities)]float64`.

Regarding the second comment, you should extract `float64(amounts) / 2` outside of the loop and this is a constant value and you could compute it beforehand.